odl_video/logging.py

```diff
@@ -0,0 +1,47 @@
+"""Configure structured logging for our application"""
+import logging
+from structlog_sentry import SentryJsonProcessor
+import structlog
+from odl_video import settings
+
+
+logging.basicConfig(level=getattr(logging, settings.LOG_LEVEL),
+                    format='%(message)s')
+
+structlog_processors_per_debug = {
+    True: [
+        structlog.stdlib.add_log_level,
+        structlog.stdlib.add_logger_name,
+        structlog.processors.StackInfoRenderer(),
+        structlog.dev.set_exc_info,
+        structlog.processors.format_exc_info,
+        structlog.processors.TimeStamper(fmt='iso'),
+        structlog.dev.ConsoleRenderer()
+    ],
+    False: [
+        structlog.stdlib.add_logger_name,
+        structlog.stdlib.add_log_level,
+        structlog.stdlib.filter_by_level,
+        structlog.processors.TimeStamper(),
+        SentryJsonProcessor(level=logging.ERROR,
+                            tag_keys=['environment', 'level', 'logger',
```

Comment:
does there need to be a tag_key for every variable used in an error message?  For instance, [this code](https://github.com/mitodl/odl-video-service/pull/865/files#diff-0da2c698f19bf424e9fdcd1e851eda69R255) adds `status` to the error message but it isn't showing up in sentry here: https://sentry.io/organizations/mit-office-of-digital-learning/issues/1615059130/?project=194353&query=is%3Aunresolved

Reply:
That `tag_keys` List in logging.py is a whitelist of tags that it will set in the Sentry issue that come from properties set in the `log.error()` call. The `status` tag was not set in the Sentry error above because it wasn't on that list, despite the code having executed `raise HttpAccessTokenRefreshError(error_msg, status=resp.status)`. I can add `status` to `tag_keys`.

If you feel that you're seeing too many tags in the errors I can recduce them. I'm noticing now that `structlog` or `sentry-sdk` is adding tags on its own. Is that OK? For instance, I didn't configure anything to make it add `trace` or `transaction`. Are you OK with the proliferation of tags?

Reply:
I'm fine with the extra tags that structlog adds on its own.

For things like `status`, can it somehow be included in the message without adding a tag for it?  I can foresee situations where I'd want to add some extra info to a particular error message without wanting to add it as a tag to the structlog setup.

For example, here is the error on CI using the old logging setup, the status shows up as variable #1 in the message: https://sentry.io/organizations/mit-office-of-digital-learning/issues/1615059130/events/c883c45d08484cec9f9cca94c89d555e/?project=194353

But on my dev instance it is not visible anywhere as far as I can tell:
https://sentry.io/organizations/mit-office-of-digital-learning/issues/1615059130/events/9b1aa3456c44436a8666715b10d2a435/?project=194353

Reply:
Sure, I'm happy to put the video ID back into the Message string of the log message, and to put variable values back into messages where you would like them.  I'll do this one now.

The cool thing about the new error (the 2nd link above) is there's a tag now named `video_hexkey` that has the video ID.  Unlike before, you can click on that tag and see all errors in the system pertaining to that particular video, instead of having to do a text search.

Reply:
(With structured logging, a similar thing will happen in Kibana. There will automagically be a field that gets indexed in Elasticsearch for that video ID. You'll be able to click on a filter icon in the record you're looking at to filter on that video ID, instead of having to go up to the search bar and search on it.)

Reply:
Changed in commit ec20c21

Reply:
I agree it's useful to have a video hexkey tag, maybe a video status tag too, but occasionally we might want to have some extra value in one particular error message that wouldn't be worth adding a tag for, so still having the option of just including it as a formatted string would be good, as in the code comment below `"Error uploading video to Youtube: %s", positional_args=(youtube_video.status,)`